--- a/unix/build-xorg-orig	2012-09-12 11:25:57.773043884 +0200
+++ b/unix/build-xorg	2012-09-12 14:31:52.059430957 +0200
@@ -176,7 +176,12 @@
     # build mesa
     echo "*** Building Mesa ***"
     pushd Mesa-*
-    ./configure ${CFGHOST} --prefix=${PREFIX} --disable-driglx-direct --with-dri-drivers=swrast --with-driver=dri --disable-glut --without-demos
+    if [ x$IS64BIT == xis64bit_yes ]; then
+      MESA_ARCH=--enable-64-bit
+    else
+      MESA_ARCH=--enable-32-bit
+    fi
+    ./configure ${CFGHOST} ${MESA_ARCH} --prefix=${PREFIX} --disable-driglx-direct --with-dri-drivers=swrast --with-driver=dri --disable-glut --without-demos
     if [ $? -ne 0 ]; then
 	echo "Failed to configure Mesa."
 	exit
@@ -193,7 +198,7 @@
     echo "*** Building xserver ***"
     pushd xorg/xserver
     autoreconf -fiv
-    XORGCFGFLAGS="--disable-dri --enable-dri2 --disable-composite --disable-xinerama --disable-xvfb --disable-xnest --disable-xorg --disable-dmx --disable-xwin --disable-xephyr --disable-kdrive --disable-config-dbus --disable-config-hal --disable-config-udev --with-sha1=libgcrypt SHA1_LIB=-lcrypto --disable-shared --enable-static ${XORGCFGFLAGS}"
+    XORGCFGFLAGS="--with-dri-driver-path=/usr/lib/tigervnc_dri --disable-dri --enable-dri2 --disable-composite --disable-xinerama --disable-xvfb --disable-xnest --disable-xorg --disable-dmx --disable-xwin --disable-xephyr --disable-kdrive --disable-config-dbus --disable-config-hal --disable-config-udev --with-sha1=libgcrypt SHA1_LIB=-lcrypto --disable-shared --enable-static ${XORGCFGFLAGS}"
     ./configure ${CFGHOST} --prefix=${PREFIX} ${XORGCFGFLAGS}
     if [ $? -ne 0 ]; then
 	echo "Failed to configure X server."
